src/main/java/com/st/utopia/cancellationservice/controller/CancellationServiceController.java

```diff
@@ -0,0 +1,51 @@
+package com.st.utopia.cancellationservice.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.st.utopia.cancellationservice.model.Ticket;
+import com.st.utopia.cancellationservice.service.TicketService;
+
+import javassist.NotFoundException;
+
+
+
+@RestController
+@RequestMapping("/Cancel")
+public class CancellationServiceController {
+	
+	@Autowired
+	private TicketService ticketservice;
+	
+	
+	@GetMapping("/FindTicket/{flightNumber}/{rowNumber}/{seat}")
+	public ResponseEntity<Ticket> getTicket(@PathVariable int flightNumber, @PathVariable int rowNumber,@PathVariable char seat) throws NotFoundException {
+		Ticket ticket = ticketservice.findTicket(flightNumber, rowNumber, seat);
+		if(ticket == null) {
+			throw new NotFoundException("ticket with flightNumber=" + flightNumber + " not found");

```

Comment:
If your method is returning a `ResponseEntity` anyway, why throw an exception rather than just returning `new ResponseEntity<>(HttpStatus.NotFound)`?

Reply:
Yes, agree. Unless you added the annotation that classifies that exceptions as a particular `ResponseEntity<>`.